modules/backendscheduler/work/job.go

```diff
@@ -118,24 +118,28 @@ func (j *Job) GetWorkerID() string {
 
 // OnBlock returns true if the job is operating on a block.
 func (j *Job) OnBlock(id string) bool {
-	for _, b := range j.GetCompactionInput() {
-		if b == id {
-			switch j.GetStatus() {
-			case tempopb.JobStatus_JOB_STATUS_RUNNING, tempopb.JobStatus_JOB_STATUS_UNSPECIFIED:
-				return true
+	switch j.GetType() {
+	case tempopb.JobType_JOB_TYPE_COMPACTION:
+		for _, b := range j.GetCompactionInput() {
+			if b == id {
+				switch j.GetStatus() {
+				case tempopb.JobStatus_JOB_STATUS_RUNNING, tempopb.JobStatus_JOB_STATUS_UNSPECIFIED:
+					return true
+				}
 			}
 		}
-	}
 
-	for _, b := range j.GetCompactionOutput() {
-		if b == id {
-			switch j.GetStatus() {
-			case tempopb.JobStatus_JOB_STATUS_RUNNING, tempopb.JobStatus_JOB_STATUS_UNSPECIFIED:
-				return true
+		for _, b := range j.GetCompactionOutput() {
+			if b == id {
+				switch j.GetStatus() {
+				case tempopb.JobStatus_JOB_STATUS_RUNNING, tempopb.JobStatus_JOB_STATUS_UNSPECIFIED:
```

Comment:
small code cleanup. these cases could be tested once on line 121:

```
status := j.GetStatus()
if status != running && status != unspecified {
  return false
}

// check other things
```

Reply:
I pushed a commit to tidy this up a bit.  It avoids some processing now, good call.